### PR TITLE
AL-370 - Roles and Permissions Service

### PR DESCRIPTION
**Pull request by mjoksa**

### CODE REVIEW

**Review by dafinkagerdjikova — APPROVED** [review]

File: src/Lykke.AlgoStore.Service.Security.Services/UserRolesService.cs [hunk 1]
```diff
@@ -266,5 +267,61 @@ public class UserRolesService : IUserRolesService
 
             await _rolesRepository.DeleteRoleAsync(role);
         }
+
+        public async Task SeedRoles(List<UserPermissionData> permissions)
```
Comment (ghost) [inline src/Lykke.AlgoStore.Service.Security.Services/UserRolesService.cs]: Rename method to SeedRolesAsync to better follow convention across the project

File: src/Lykke.AlgoStore.Service.Security.Services/UserPermissionsService.cs [hunk 2]
```diff
@@ -143,5 +143,52 @@ public class UserPermissionsService : IUserPermissionsService
 
             return userRoles.Any(x => x.Permissions.Any(y => y.Id == permissionId));
         }
+
+        public async Task SeedPermissions(List<UserPermissionData> permissions)
```
Comment (ghost) [inline src/Lykke.AlgoStore.Service.Security.Services/UserPermissionsService.cs]: Rename method to SeedPermissions to better follow convention across the project

File: client/Lykke.AlgoStore.Service.Security.Client/Lykke.AlgoStore.Service.Security.Client.xml [hunk 3]
```diff
@@ -0,0 +1,1405 @@
+<?xml version="1.0"?>
```
Comment (ghost) [inline client/Lykke.AlgoStore.Service.Security.Client/Lykke.AlgoStore.Service.Security.Client.xml]: Perhaps this file should be removed, as was requested during the MEA review

File: src/Lykke.AlgoStore.Service.Security.Core/Utils/FuncExtensions.cs [hunk 4]
```diff
@@ -7,7 +7,7 @@ namespace Lykke.AlgoStore.Service.Security.Core.Utils
 {
     public static class FuncExtensions
```
Comment (ghost) [inline src/Lykke.AlgoStore.Service.Security.Core/Utils/FuncExtensions.cs]: Perhaps this should be renamed to LogExtensions to better represent its functionality

File: src/Lykke.AlgoStore.Service.Security.Core/Utils/FuncExtensions.cs [hunk 5]
```diff
@@ -7,7 +7,7 @@ namespace Lykke.AlgoStore.Service.Security.Core.Utils
...
     {
-        public static async Task LogElapsedTime(this Func<Task> action, ILog log, string clientId)
+        public static async Task LogElapsedTime(this ILog log, string clientId, Func<Task> action)
```
Comment (ghost) [inline src/Lykke.AlgoStore.Service.Security.Core/Utils/FuncExtensions.cs]: This should be renamed to LogElapsedTimeAsync to better follow convention around the project

File: src/Lykke.AlgoStore.Service.Security.Core/Services/IUserRolesService.cs [hunk 6]
```diff
@@ -10,11 +10,12 @@ public interface IUserRolesService
         Task<UserRoleData> GetRoleByIdAsync(string roleId);
         Task<List<UserRoleData>> GetRolesByClientIdAsync(string clientId);
         Task<List<AlgoStoreUserData>> GetAllUsersWithRolesAsync();
-        Task<AlgoStoreUserData> GeyUserByIdWithRoles(string clientId);
-        Task AssignRoleToUser(UserRoleMatchData data);
+        Task<AlgoStoreUserData> GeyUserByIdWithRolesAsync(string clientId);
```
Comment (ghost) [inline src/Lykke.AlgoStore.Service.Security.Core/Services/IUserRolesService.cs]: Gey -> Get

**Review by ghost — APPROVED** [review]

**Pull request merged**